test/test_distributions.py

```diff
@@ -1443,6 +1456,113 @@ def test_normal_sample(self):
+                                 (mean, cov_factor, cov_diag))
+        self._gradcheck_log_prob(LowRankMultivariateNormal,
+                                 (mean_multi_batch, cov_factor, cov_diag))
+        self._gradcheck_log_prob(LowRankMultivariateNormal,
+                                 (mean_multi_batch, cov_factor_batched, cov_diag_batched))
+
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_lowrank_multivariate_normal_log_prob(self):
+        mean = torch.randn(3, requires_grad=True)
+        cov_factor = torch.randn(3, 1, requires_grad=True)
+        cov_diag = torch.tensor(torch.randn(3).abs(), requires_grad=True)
+        cov = cov_factor.matmul(cov_factor.t()) + cov_diag.diag()
+
+        # check that logprob values match scipy logpdf,
+        # and that covariance and scale_tril parameters are equivalent
+        dist1 = LowRankMultivariateNormal(mean, cov_factor, cov_diag)
+        ref_dist = scipy.stats.multivariate_normal(mean.detach().numpy(), cov.detach().numpy())
+
+        x = dist1.sample((10,))
+        expected = ref_dist.logpdf(x.numpy())
+
+        self.assertAlmostEqual(0.0, np.mean((dist1.log_prob(x).detach().numpy() - expected)**2), places=3)
+
+        # Double-check that batched versions behave the same as unbatched
+        mean = torch.randn(5, 3, requires_grad=True)
+        cov_factor = torch.randn(5, 3, 2, requires_grad=True)
+        cov_diag = torch.tensor(torch.randn(5, 3).abs(), requires_grad=True)
+
+        dist_batched = LowRankMultivariateNormal(mean, cov_factor, cov_diag)
+        dist_unbatched = [LowRankMultivariateNormal(mean[i], cov_factor[i], cov_diag[i])
+                          for i in range(mean.size(0))]
+
+        x = dist_batched.sample((10,))
+        batched_prob = dist_batched.log_prob(x)
+        unbatched_prob = torch.stack([dist_unbatched[i].log_prob(x[:, i]) for i in range(5)]).t()
+
+        self.assertEqual(batched_prob.shape, unbatched_prob.shape)
+        self.assertAlmostEqual(0.0, (batched_prob - unbatched_prob).abs().max(), places=3)
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    def test_lowrank_multivariate_normal_sample(self):
+        set_rng_seed(0)  # see Note [Randomized statistical tests]
+        mean = torch.randn(5, requires_grad=True)
+        cov_factor = torch.randn(5, 1, requires_grad=True)
+        cov_diag = torch.tensor(torch.randn(5).abs(), requires_grad=True)
+        cov = cov_factor.matmul(cov_factor.t()) + cov_diag.diag()
+
+        self._check_sampler_sampler(LowRankMultivariateNormal(mean, cov_factor, cov_diag),
+                                    scipy.stats.multivariate_normal(mean.detach().numpy(), cov.detach().numpy()),
+                                    'LowRankMultivariateNormal(loc={}, cov_factor={}, cov_diag={})'
+                                    .format(mean, cov_factor, cov_diag), multivariate=True)
+
+    def test_lowrank_multivariate_normal_properties(self):
```

Comment:
@fehiepsi This tests agreement between `LowRankMultivariateNormal` and `MultivariateNormal` moments, but does not actually tests moments of the generated samples. Could you add a test that checks agreement between `.rsample`, and `(.mean, .variance)` with say 10000 samples? Something like this:
```py
def test_lowrank_multivariate_normal_moments(self):
    mean = torch.randn(5, requires_grad=True)
    cov_factor = torch.randn(5, 2, requires_grad=True)
    cov_diag = torch.tensor(torch.randn(5).abs(), requires_grad=True)
    d = LowRankMultivariateNormal(mean, cov_factor, cov_diag)
    samples = d.rsample((100000,))
    empirical_mean = samples.mean(0)
    assert_equal(d.mean, empirical_mean, prec=0.01)
    empirical_var = samples.var(0)
    assert_equal(d.variance, empirical_var, prec=0.01)
```

Reply:
You are right! Indeed, we also have no such test for multivariate_normal. So I add your test for both cases. Note that the `prec` for variance is higher than 0.01 (0.02 for `lowrank` and 0.05 for `full`). Do you think that it is fine?

Reply:
Those precisions look fine for this PR, but could you verify that the tests also pass at precision 0.01 by temporarily increasing sample count to e.g. `int(1e7)`?

Reply:
Yes, it passed in both cases. I have verified it. :)